### PR TITLE
Add missing semi-colon

### PR DESCRIPTION
**Pull request by sanjayankur31**
Tiny syntax fix.

### CODE REVIEW

File: arbor/include/arbor/simd/neon.hpp [hunk 1]
```diff
@@ -626,7 +626,7 @@ struct neon_double2 : implbase<neon_double2> {
     }
 };
 
-}  // namespace detail
+};  // namespace detail
```
Comment (bcumming) [inline arbor/include/arbor/simd/neon.hpp]: Semi-colons are not required at the end of namespace definitions. To the extent of my knowledge, the only time a semi-colon is required after a `}` is at the end of a `class`, `struct`, or `union` definition.

Comment (sanjayankur31) [reply arbor/include/arbor/simd/neon.hpp]: It's a struct definition. The matching brace is on line 190 at  ```struct neon_double2 : implbase<neon_double2> {```

Comment (sanjayankur31) [reply arbor/include/arbor/simd/neon.hpp]: Either the comments are off or matching brackets are missing: vim tells me that `namespace detail` ends on line 646 which says `// namespace simd`, and nothing matches the bracket for `// namespace arb` on line 647.

Comment (noraabiakar) [reply arbor/include/arbor/simd/neon.hpp]: A closing bracket was accidentally deleted in #925 line 489

Comment (sanjayankur31) [reply arbor/include/arbor/simd/neon.hpp]: Ah, then putting that back would be the correct fix. I'll close this and let you fix it up properly.

**Pull request closed without merging**